### PR TITLE
Unicode server

### PR DESCRIPTION
**Pull request by amn41**
addresses issue #44 

@plauto I would still have a test which includes the server, e.g. 

`requests.post(http_server + "/parse", json={"q": u"hello ńöñàśçií"})` in `test_post_parse()`

and you're right that entities should be a **list**, but with the start and end of each entity specified, e.g. 
```json
      {
        "text": "show me chinese restaurants",
        "intent": "restaurant_search",
        "entities": [
          {
            "start": 8,
            "end": 15,
            "value": "chinese",
            "entity": "cuisine"
          }
        ]
      }```

### CODE REVIEW

File: src/emulators/wit.py [hunk 1]
```diff
@@ -4,17 +4,21 @@ def __init__(self):
 
     def normalise_request_json(self, data):
         _data = {}
+        print data
```
Comment (amn41) [inline src/emulators/wit.py]: remove printf

File: _pytest/test_tokenizers.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ def tokenize_sentence(sentence, expected_result, language):
         assert tk.tokenize(sentence) == expected_result
```
Comment (amn41) [inline _pytest/test_tokenizers.py]: please also add a nonascii test for whitespace tokenizer

File: _pytest/test_emulators.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ def test_wit_request():
 def test_wit_response():
     from rasa_nlu.emulators.wit import WitEmulator
     em = WitEmulator()
-    data = {"text": "I want italian food", "intent": "inform", "entities": {"cuisine": "italian"}}
+    data = {"text": "I want italian food", "intent": "inform", "entities": [{"entity": "cuisine", "value": "italian"}]}
```
Comment (amn41) [inline _pytest/test_emulators.py]: please also include the `"start"` and `"end"` parts of the entities

File: _pytest/test_server.py [hunk 4]
```diff
@@ -48,22 +57,41 @@ def test_status(http_server):
...
-    req = requests.get(http_server + "/parse?q=hello")
-    expected = [{"entities": {}, "confidence": None, "intent": "greet", "_text": "hello"}]
-    assert req.status_code == 200 and req.json() == expected
+    tests = [
+        ResponseTest(
+            "/parse?q=hello",
+            [{u"entities": {}, u"confidence": None, u"intent": u"greet", u"_text": u"hello"}]
+        ),
+        ResponseTest(
+            "/parse?q=hello ńöñàśçií",
```
Comment (amn41) [inline _pytest/test_server.py]: why isn't this a unicode obj instead of a string?

File: _pytest/test_server.py [hunk 5]
```diff
@@ -48,22 +57,41 @@ def test_status(http_server):
...
 
 def test_post_parse(http_server):
-    req = requests.post(http_server + "/parse", json={"q": "hello"})
-    expected = [{"entities": {}, "confidence": None, "intent": "greet", "_text": "hello"}]
-    assert req.status_code == 200 and req.json() == expected
+    tests = [
+        ResponseTest(
+            "/parse",
+            [{u"entities": {}, u"confidence": None, u"intent": u"greet", u"_text": u"hello"}],
+            payload={"q": "hello"}
```
Comment (amn41) [inline _pytest/test_server.py]: in the `POST` parse method the json should be e.g. `{"text":"hello"}`

File: _pytest/test_server.py [hunk 6]
```diff
@@ -48,22 +57,41 @@ def test_status(http_server):
...
+        ),
+        ResponseTest(
+            "/parse",
+            [{u"entities": {}, u"confidence": None, u"intent": u"greet", u"_text": u"hello ńöñàśçií"}],
+            payload={"q": "hello ńöñàśçií"}
```
Comment (amn41) [inline _pytest/test_server.py]: same here, q ->text

File: src/emulators/wit.py (hunk 1 above)
Comment (plauto) [inline src/emulators/wit.py]: sure thing!

File: _pytest/test_emulators.py (hunk 3 above)
Comment (plauto) [inline _pytest/test_emulators.py]: I checked how wit formats the response and those keys are not present in the original one. Do you want me to add them anyway?

File: _pytest/test_server.py (hunk 5 above)
Comment (plauto) [inline _pytest/test_server.py]: I checked the documentation, and actually there we use q...did we change to text at some point? If so I can fix it back in no time :)

File: _pytest/test_emulators.py (hunk 3 above)
Comment (amn41) [inline _pytest/test_emulators.py]: but `data` is what's produced by rasa correct? `norm` should have the wit format

File: _pytest/test_server.py (hunk 5 above)
Comment (amn41) [inline _pytest/test_server.py]: that's my mistake. I mean for simplicity I suppose we could use q as well. Just so long as it's consistent everywhere

File: _pytest/test_emulators.py (hunk 3 above)
Comment (plauto) [inline _pytest/test_emulators.py]: true! My bad. Okay, I am going to add them right away
